### ISSUE TITLE
Weather no longer works

### ISSUE THREAD

**Issue by jamerst** [marcopixel/SysDash#44]
As of 3rd January 2019, query.yahooapis.com for Yahoo weather is deprecated, so the weather fetching is broken.

[This](https://developer.yahoo.com/weather/) page gives details of the replacement system.

**Comment by emigdio821** [comment]
Same here.

**Comment by paulternate** [comment]
Yep, weather link is broken now.

**Comment by Acidinmyfridge** [comment]
Dito, weather link does not work anymore.

**Comment by crypt3x-1988** [comment]
I've ran into the same issue.

Fiddled around in the weather.ini File and changed the WebParser to use the weather.com API instead.

If you want to do that on your own, you'll need to get the Location Code for your City and change that in the Url! Currently its set to Berlin/Germany

The units are set to metric "units=m" if you want imperial units instead change that to "units=f"



`; Measure to get the weather data and properly parse it
[MeasureWeather]
Measure=Plugin
Plugin=WebParser
Url=http://wxdata.weather.com/wxdata/weather/local/GMXX0007?cc=*&unit=m
RegExp=(?siU)<weather ver="(.*)">.*<ut>(.*)</ut>.*<ud>(.*)</ud>.*<us>(.*)</us>.*<up>(.*)</up>.*<ur>(.*)</ur>.*<dnam>(.*)</dnam>.*<lat>(.*)</lat>.*<lon>(.*)</lon>.*<sunr>(.*)</sunr>.*<suns>(.*)</suns>.*<zone>(.*)</zone>.*<tmp>(.*)</tmp>.*<flik>(.*)</flik>.*<t>(.*)</t>.*<icon>(.*)</icon>.*<bar>.*<r>(.*)</r>.*<d>(.*)</d>.*</bar>.*<wind>.*<s>(.*)</s>.*<gust>(.*)</gust>.*<d>(.*)</d>.*<t>(.*)</t>.*</wind>.*<hmid>(.*)</hmid>.*<vis>(.*)</vis>.*<moon>.*<icon>(.*)</icon>.*<t>(.*)</t>.*</moon>

UpdateRate=30
FinishAction=[!Update] [!Update]
[MeasureCurrentCity]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=7
[MeasureCurrentCode]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=16
Substitute="":"na"
[MeasureCurrentTemp]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=13
Substitute="":"N/A"
[MeasureCurrentDesc]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=15
Substitute="Scattered":""
`

**Comment by r0tciv9** [comment]
The above worked for me. I found my location using https://weather.codes/
Below is the exact contents of my weather.ini file if you want to be able to directly copy it over; just replace YOURCODE in ```Url=http://wxdata.weather.com/wxdata/weather/local/YOURCODE?cc=*&unit=f```

```
[Rainmeter]
Group=SysDash | Weather
Update=10000
AccurateText=1
BackgroundMode=2
SolidColor=0,0,0,1

; = IMPORTANT =
; = For configuring the rainmeter skin, right-click the visualizer and click on "Open settings".
; = You can also open the variables.ini file located in:
; = "My Documents\Rainmeter\Skins\SysDash\@Resources"

; Small context menu when you right-click the skin
ContextTitle=" Open settings"
ContextAction=[!ActivateConfig "#ROOTCONFIG#\Settings" "general.ini"]
ContextTitle2=" Open variables file"
ContextAction2=["#@#variables.ini"]

[Metadata]
Name=SysDash Dashboard
Author=marcopixel
License=MIT License
Information=An minimalistic, still stylish dashboard-like skin with modular components.

[Variables]
; Includes the variables used for the skin.
@include=#@#variables.ini
@include2=#@#include\MeterStyles.inc

; Measure to get the weather data and properly parse it
[MeasureWeather]
Measure=Plugin
Plugin=WebParser
Url=http://wxdata.weather.com/wxdata/weather/local/USCA0087?cc=*&unit=f
RegExp=(?siU)<weather ver="(.*)">.*<ut>(.*)</ut>.*<ud>(.*)</ud>.*<us>(.*)</us>.*<up>(.*)</up>.*<ur>(.*)</ur>.*<dnam>(.*)</dnam>.*<lat>(.*)</lat>.*<lon>(.*)</lon>.*<sunr>(.*)</sunr>.*<suns>(.*)</suns>.*<zone>(.*)</zone>.*<tmp>(.*)</tmp>.*<flik>(.*)</flik>.*<t>(.*)</t>.*<icon>(.*)</icon>.*<bar>.*<r>(.*)</r>.*<d>(.*)</d>.*</bar>.*<wind>.*<s>(.*)</s>.*<gust>(.*)</gust>.*<d>(.*)</d>.*<t>(.*)</t>.*</wind>.*<hmid>(.*)</hmid>.*<vis>(.*)</vis>.*<moon>.*<icon>(.*)</icon>.*<t>(.*)</t>.*</moon>

UpdateRate=30
FinishAction=[!Update] [!Update]
[MeasureCurrentCity]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=7
[MeasureCurrentCode]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=16
Substitute="":"na"
[MeasureCurrentTemp]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=13
Substitute="":"N/A"
[MeasureCurrentDesc]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=15
Substitute="Scattered":""

[MeterWeatherIcon]
Meter=Image
MeasureName=MeasureCurrentCode
Path=#@#images\weather
X=(#Margin#*#Scale#)
Y=(20*#Scale#)
W=(40*#Scale#)
H=(40*#Scale#)
Padding=0,0,0,(20*#Scale#)
ImageTint=#MainColor#
ImageCrop=-30,-30,61,61,5

[MeterWeatherTempText]
Meter=String
MeasureName=MeasureCurrentTemp
MeterStyle=StyleValue
X=(15*#Scale#)R
Y=((80/2)*#Scale#)
Postfix=[\x00B0]

[MeterWeatherCityText]
Meter=String
MeasureName=MeasureCurrentCity
MeterStyle=StyleSecondary
X=((#Width#+#Margin#)*#Scale#)
Y=(30*#Scale#)
InlineSetting4=Color | #FontColor#,255
[MeterWeatherDescText]
Meter=String
MeasureName=MeasureCurrentDesc
MeterStyle=StyleSecondary
X=0r
Y=0R
```

**Comment by bennony** [comment]
Works with me unfortunately not quite (copied from (r0tciv9), the weather icon is displayed, and the number of degrees, but not place, the degree sign is not synonymous, and the temperature is displayed in black, unfortunately I do not know how I do it can change to white!

Even less works when I copy from crypt3x-1988, you can not do a lot wrong or right? Copy and paste his place (GMXX0015) ready?

sorry for my google English, only speak German!

**Comment by dkolb** [comment]
I made the same changes as @r0tciv9 but also updated the `MeasureCurrentCity` with the following regex substitutions to trim down on locations with multiple city names and to strip out the state and country portions.

```ini
[MeasureCurrentCity]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=10
RegExpSubstitute=1
Substitute="^(.*), (.*), (.*)$":"\1","^(.*)\/.*$":"\1"
```

**Comment by bennony** [comment]
so läuft es bei mir nun OK 

`[Rainmeter]
Group=SysDash | Weather
Update=10000
AccurateText=1
BackgroundMode=2
SolidColor=0,0,0,1

; = IMPORTANT =
; = For configuring the rainmeter skin, right-click the visualizer and click on "Open settings".
; = You can also open the variables.ini file located in:
; = "My Documents\Rainmeter\Skins\SysDash\@Resources"

; Small context menu when you right-click the skin
ContextTitle=" Open settings"
ContextAction=[!ActivateConfig "#ROOTCONFIG#\Settings" "general.ini"]
ContextTitle2=" Open variables file"
ContextAction2=["#@#variables.ini"]

[Metadata]
Name=SysDash Dashboard
Author=marcopixel
License=MIT License
Information=An minimalistic, still stylish dashboard-like skin with modular components.

[Variables]
; Includes the variables used for the skin.
@include=#@#variables.ini
@include2=#@#include\MeterStyles.inc

; Measure to get the weather data and properly parse it
[MeasureWeather]
Measure=Plugin
Plugin=WebParser
Url=http://wxdata.weather.com/wxdata/weather/local/GMXX0015?cc=*&unit=m
RegExp=(?siU)<weather ver="(.*)">.*<ut>(.*)</ut>.*<ud>(.*)</ud>.*<us>(.*)</us>.*<up>(.*)</up>.*<ur>(.*)</ur>.*<dnam>(.*)</dnam>.*<lat>(.*)</lat>.*<lon>(.*)</lon>.*<sunr>(.*)</sunr>.*<suns>(.*)</suns>.*<zone>(.*)</zone>.*<tmp>(.*)</tmp>.*<flik>(.*)</flik>.*<t>(.*)</t>.*<icon>(.*)</icon>.*<bar>.*<r>(.*)</r>.*<d>(.*)</d>.*</bar>.*<wind>.*<s>(.*)</s>.*<gust>(.*)</gust>.*<d>(.*)</d>.*<t>(.*)</t>.*</wind>.*<hmid>(.*)</hmid>.*<vis>(.*)</vis>.*<moon>.*<icon>(.*)</icon>.*<t>(.*)</t>.*</moon>

UpdateRate=100
FinishAction=[!Update] [!Update]
[MeasureCurrentCity]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=7
[MeasureCurrentCode]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=16
Substitute="":"na"
[MeasureCurrentTemp]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=13
Substitute="":"N/A"
[MeasureCurrentDesc]
Measure=Plugin
Plugin=WebParser
URL=[MeasureWeather]
StringIndex=15
Substitute="Scattered":""

[MeterWeatherIcon]
Meter=Image
MeasureName=MeasureCurrentCode
Path=#@#images\weather
X=(#Margin#*#Scale#)
W=(40*#Scale#)
H=(40*#Scale#)
Padding=0,(15*#Scale#),0,(15*#Scale#)
ImageTint=#MainColor#

[MeterWeatherTempText]
Meter=String
MeasureName=MeasureCurrentTemp
MeterStyle=StyleValue
X=(15*#Scale#)R
Y=((70/2)*#Scale#)
StringAlign=LeftCenter
InlineSetting=Face | #Font#
InlineSetting2=Weight | 100
InlineSetting3=Size | (30*#Scale#)
InlineSetting4=Color | #FontColor#,235
AntiAlias=1
Postfix=[\x00B0]

[MeterWeatherCityText]
Meter=String
MeasureName=MeasureCurrentCity
MeterStyle=StyleSecondary
X=((#Width#+#Margin#)*#Scale#)
Y=(15*#Scale#)
W=((#Width#-120)*#Scale#)
H=(45*#Scale#)
InlineSetting=Face | #Font#
InlineSetting2=Weight | 200
InlineSetting3=Size | (14*#Scale#)
InlineSetting4=Color | #FontColor#,150
AntiAlias=1
StringAlign=Right
ClipString=2
Text=%1
`

**Comment by Solarwind2** [comment]
Anyway to display the weather the the city locale language ? Would like to see my weather in my language

**Comment by marcopixel** [comment]
Will be fixed in the future, check #56 for more information.
